vms/platformvm/block/builder/builder_test.go

```diff
@@ -320,6 +320,115 @@ func TestGetNextStakerToReward(t *testing.T) {
 	}
 }
 
+func TestBuildBlockShouldReward(t *testing.T) {
+	require := require.New(t)
+
+	env := newEnvironment(t)
+	env.ctx.Lock.Lock()
+	defer func() {
+		require.NoError(shutdownEnvironment(env))
+		env.ctx.Lock.Unlock()
+	}()
+
+	var (
+		now    = env.backend.Clk.Time()
+		nodeID = ids.GenerateTestNodeID()
+
+		defaultValidatorStake = 100 * units.MilliAvax
+		validatorStartTime    = now.Add(2 * txexecutor.SyncBound)
+		validatorEndTime      = validatorStartTime.Add(360 * 24 * time.Hour)
+	)
+
+	// Create a valid [AddValidatorTx]
+	tx, err := env.txBuilder.NewAddValidatorTx(
+		defaultValidatorStake,
+		uint64(validatorStartTime.Unix()),
+		uint64(validatorEndTime.Unix()),
+		nodeID,
+		preFundedKeys[0].PublicKey().Address(),
+		reward.PercentDenominator,
+		[]*secp256k1.PrivateKey{preFundedKeys[0]},
+		preFundedKeys[0].PublicKey().Address(),
+	)
+	require.NoError(err)
+	txID := tx.ID()
+
+	// Issue the transaction
+	require.NoError(env.network.IssueTx(context.Background(), tx))
+	require.True(env.mempool.Has(txID))
+
+	// Build and accept a block with the tx
+	blk, err := env.Builder.BuildBlock(context.Background())
+	require.NoError(err)
+	require.IsType(&block.BanffStandardBlock{}, blk.(*blockexecutor.Block).Block)
```

Comment:
Although this is a direct replacement for the "should reward" test, the "has decision txs" test functionality is also replicated here since we build a block ^